### ISSUE TITLE
Cannot finish repetitive `every 3 days` tasks from Todoist

### ISSUE THREAD

**Issue by ai** [alainm23/planify#1077]
**Describe the bug**

In Todoist I have task with `every 3 days`. When I finish it in the app, the app shows `next day: today` and task is still in the list.

**To Reproduce**
1. Add Todoist account
3. Create a task with `every 3 days` in text
4. Open Planify and click task’s checkbox
5. The tasks will not be finished, and `every 3 days` mark will be removed from Todoist.

**Expected behavior**

Hide task from the list and show it again in 3 days.

**Screenshots**

![Screenshot from 2023-12-16 14-00-01](https://github.com/alainm23/planify/assets/19343/fe106559-066f-4571-a6c8-26eb1788cf16)

(Screenshot made in 16 of December)

**Desktop (please complete the following information):**
 - OS or DE: GNOME 44
 - Version: Planify 4.2.1

**Comment by dmytrodubinin** [comment]
I have similar problem with repeated tasks. When complete repeated task in Planify nothing happens in app but  in Todoist this task repeat become disabled

**Comment by alainm23** [comment]
Thanks for reporting the problem, the recurring tasks in Toodoist are not yet compatible with Planify, Planify handles its own system of recurring tasks, it works fine if the programs in Planify.

**Comment by alainm23** [comment]
However, we could make a basic compatibility for the moment and for English.

**Comment by ai** [comment]
> for English

Are we talking about repetitive task creation? Or Todoist store repetitive pattern as a text description in natural language (`"every 3 days"`)?

**Comment by alainm23** [comment]
@ai  Exactly, Todoist handles it in a natural language, hence the complexity, we could give compatibility to the most basic (for now) and add more along the way.

- Every day
- Every month
- Every year
- Every 2 days
- Every 2 months

**Comment by ai** [comment]
I personally, think that English is better than nothing. Can we show a warning that we can complete this task since its repetitive pattern can’t be understood (try to rewrite it to English)?

**Comment by alainm23** [comment]
Ideally, the user should never be able to enter a repeating pattern, this should be generated automatically. This will be the flow.

If you create a repeating task in Todoist and sync with Planify, Planify should be able to read this repeating task and make it work and the same from Planify to Todoist. The only problem would be if the user created a very complex one, for example (every 14 days 7am starting 1 Nov), in that case we could show an alert indicating that it is not possible to make it work at the moment.

**Comment by tmrlvi** [comment]
I was playing around with how Todoist parses and reports repeating pattern. While if the user enters the pattern using the dialog it creates somewhat standardize output, the more common approach (and the approach Todoist prefer the users to use) is to write the pattern as part of the task (e.g. `Run every weekend`).

They have an internal library that parses and serializes the pattern (`@doist/dateist`), but it is not very clean - it does handle typos (e.g. `every Firday` becomes `every friday`) but does not normalize or checks the output (e..g `every Sun, Firday and Friday` becomes `every Sun, friday, Friday`). 

In general, the structure for recurrence is as follows
```
<every|every!> <recurrence> [time] [ending <end date>]
```
Where:
  -  `every!` implies that repetition is according to when the task was done (e.g. `every! week`  means schedule the next event a week after the current is done (not supported by Planify)
  - `<recurrence>` can be one of the following:
    - `[interval] minute|hour|day|weekday|week|month|year` (could also be in plural). Interval could be a number or `other`. 
      - If without interval (except `other`) could be also the day name (e.g., `Friday`, `friday`, `fri`, `Fri`). With interval, and a single day name, this becomes relative date.
    - `weekend` (seems can't have interval)
    - Date in the any form (seems all `09/06`, `09-06`, `Sept 6th`, `September 6th`, `Sept 6`, `September 6` are supported and remain the same after parsing)
    - Relative day in the month `2nd Monday` or  `2 Monday` (i.e. the second monday every month)
  - `[time]` is optional. Could be hourly in the format `12pm` or `10am` (could be parsed with `%I%P`) or exact, in the format `@ 12:30` or `@ 15:00`. When time exists, it also appears in the date field.
  - `<end time>` is usually `%Y-%m-%d` (e.g. `2025-12-10`), but I saw examples of `Jun 6th` etc. I'm not sure how it decides between them.

There are also patterns that Todoist doesn't parse correctly but still keeps it in the string (e.g. `every 2nd Friday and Monday` kept as is in the string, but in the UI shows as every Tuesday and Friday 🤷 )

To explore the options without syncing all the time, you can `Inspect Element` on the highlighted recurrence pattern in the Todoist web UI. The serialized patterns appears in the `data-match-id` attribute, as shown below.
![Image](https://github.com/user-attachments/assets/a452134e-d0db-4e25-99bf-fcdf46e306c8)

You can try [example of patterns Todoist claim to understand](https://www.todoist.com/help/articles/introduction-to-recurring-dates-YUYVJJAV).

**Comment by tmrlvi** [comment]
**tl;dr** - seems Todoist parse and re-serialize the recurring pattern into natural language (both if input is in natural language or in dialog). They are not 100% consistent in parsing, so I don't think Planify needs to be consistent as well. 

I suggest Planify supports the interval pattern (including `other` and `weekday`), abosulte date pattern, and the relative pattern. As well, Planify should probably support the `ending` part.

We can ignore the time (or use it to validate the date) and we can ignore `every!` (as not supported)

Todoist seem to do so with a collection of Regexes (in the `@doist/dateist` package), so maybe Planify should do it this way too.

**Comment by korintheninth** [comment]
I believe this has been fixed with #2530
